In [2]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns 
from pathlib import Path
data_path = Path("../src")
dataset = Path(data_path / "AnnualCauseDeathNumbers.csv")
data = pd.read_csv(dataset)

### En este apartado, estudiaremos las muertes causadas por enfermedades que surgen en las personas.

### En mi opinión, las enfermedades tienden a tener un mayor porcentaje de mortalidad en lugares dónde es díficil acceder a una cura.

### Algunos motivos pueden ser porque la sanidad en ese país no sea pública y el precio de la medicina se encuentre muy alto, o porque el nivel de pobreza del país sea muy elevado.

In [6]:
bio = ['Entity','Year','Meningitis fatalities','Parkinson fatalities','Malaria fatalities','HIV/AIDS fatalities','Tuberculosis fatalities',"Diarrheal disease fatalities","Neoplasm fatalities","Diabetes fatalities","Chronic kidney fatalities","Digestive disease fatalities","Acute hepatitis fatalities","Measles fatalities"]
bio


['Entity',
 'Year',
 'Meningitis fatalities',
 'Parkinson fatalities',
 'Malaria fatalities',
 'HIV/AIDS fatalities',
 'Tuberculosis fatalities',
 'Diarrheal disease fatalities',
 'Neoplasm fatalities',
 'Diabetes fatalities',
 'Chronic kidney fatalities',
 'Digestive disease fatalities',
 'Acute hepatitis fatalities',
 'Measles fatalities']

### Comenzaremos realizando un filtro a la fecha más cercana al día de hoy en el dataset, es decir, 2019.

In [14]:
b2019 = data.query('Year == 2019')
b2019[bio].sample(frac=0.8)

Entity  Year  Meningitis fatalities  Parkinson fatalities  \
6449    Uzbekistan  2019                    175                   544   
269      Argentina  2019                    453                  3043   
389     Azerbaijan  2019                     62                   338   
5549   South Korea  2019                     77                  3730   
1349  Cote dIvoire  2019                   2259                   383   
...            ...   ...                    ...                   ...   
5669     Sri Lanka  2019                    297                   858   
4829         Qatar  2019                      5                    19   
5969      Thailand  2019                   1037                  4082   
4139     Nicaragua  2019                     44                   218   
6059         Tonga  2019                      5                     5   

      Malaria fatalities  HIV/AIDS fatalities  Tuberculosis fatalities  \
6449                   0                  585                     2445   
269                    0                 1777                      727   
389                    0                   37                      636   
5549                   2                  138                     3154   
1349               29398                13638                     7086   
...                  ...                  ...                      ...   
5669                   0                   38                      873   
4829                   0                    4                        9   
5969                   1                17215                     7775   
4139                   3                  561                      143   
6059                   0                    2                        5   

      Diarrheal disease fatalities  Neoplasm fatalities  Diabetes fatalities  \
6449                           161                22338                 8154   
269                            757                85209                10762   
389                            114                11913                 2150   
5549                           991               104176                12862   
1349                          7489                11587                 3513   
...                            ...                  ...                  ...   
5669                          1019                18724                13273   
4829                             6                  831                  345   
5969                          6211               115544                19570   
4139                           195                 4429                 1904   
6059                             3                  127                   82   

      Chronic kidney fatalities  Digestive disease fatalities  \
6449                       3971                         15019   
269                       15665                         17792   
389                        1386                          3945   
5549                       7979                         17385   
1349                       3434                          6623   
...                         ...                           ...   
5669                       5560                          5350   
4829                        133                           221   
5969                      27446                         32635   
4139                       3475                          2237   
6059                         32                            37   

      Acute hepatitis fatalities  Measles fatalities  
6449                         130                   0  
269                           36                   0  
389                           24                   1  
5549                          28                   0  
1349                         220                 221  
...                          ...                 ...  
5669                          23                  35  
4829                           2                   0  
5969     

### Escogemos una muestra aleatoria de países para comparar algunos datos.

### A primera vista, la columna que más llama la atención es la de muertes por malaria.

### En 7 de los 8 países podemos encontrar que no hay muertes, o casi ninguna, por esta enfermedad. En cambio, en Costa de Marfil hay 29398 muertos.

### Si investigamos sobre la malaria, encontramos que es una enfermedad infecciosa producida por parásitos y transmitida por mosquitos hembra. 

### La malaria es una enfermedad que lleva infectando a los humanos durante más de 50000 años. Era una enfermedad endémica africana hasta que se distribuyo al continente americano con el tráfico de esclavos en el siglo XV.


### La cura de la Malaria está compuesta de un conjunto de medicamentos y antibióticos. 

### Por otro lado, la prevención se lleva a cabo con una vacuna, la eliminación de mosquitos y la prevención de picaduras.

### Para comprobar algunas suposiciones, vamos a recolectar datos sobre países africanos como Costa de Marfil.

In [19]:
af = b2019.query('Entity in ["Algeria","Morocco","Zimbabwe","South Africa","Kenya","Cote dIvoire","Botswana","Guinea","Congo","Democratic Republic of Congo","Gabon","Ghana","","Central African Republic","Madagascar","Mozambique","Nigeria","Senegal","Sierra Leona","Somalia",""]')
L = ['Entity','Malaria fatalities']
af[L]


Entity  Malaria fatalities
119                        Algeria                   0
779                       Botswana                  14
1079      Central African Republic                4248
1259                         Congo                2102
1349                  Cote dIvoire               29398
1499  Democratic Republic of Congo               57160
2189                         Gabon                 715
2309                         Ghana               21597
2489                        Guinea               11336
3029                         Kenya                8049
3419                    Madagascar                5920
3899                       Morocco                   0
3929                    Mozambique               20528
4199                       Nigeria              191106
5219                       Senegal                6740
5459                       Somalia                3899
5489                  South Africa                 464
6839                      Zimbabwe                2068